crates/vision/src/ransac.rs

```diff
@@ -24,34 +24,42 @@ impl Ransac {
 }
 
 impl Ransac {
-    pub fn next_line(&mut self, iterations: usize, maximum_distance: f32) -> RansacResult {
+    pub fn next_line(
+        &mut self,
+        iterations: usize,
+        maximum_score_distance: f32,
+        maximum_inclusion_distance: f32,
+    ) -> RansacResult {
         if self.unused_points.len() < 2 {
             return RansacResult {
                 line: None,
                 used_points: vec![],
             };
         }
-        let maximum_distance_squared = maximum_distance * maximum_distance;
+        let maximum_score_distance_squared = maximum_score_distance * maximum_score_distance;
+        let maximum_inclusion_distance_squared =
+            maximum_inclusion_distance * maximum_inclusion_distance;
         let best_line = (0..iterations)
             .map(|_| {
                 let mut points = self
                     .unused_points
                     .choose_multiple(&mut self.random_number_generator, 2);
                 let line = Line(*points.next().unwrap(), *points.next().unwrap());
-                let score = self
+                let score: f32 = self
                     .unused_points
                     .iter()
-                    .filter(|point| {
-                        line.squared_distance_to_point(**point) <= maximum_distance_squared
+                    .filter(|&point| {
+                        line.squared_distance_to_point(*point) <= maximum_score_distance_squared
                     })
-                    .count();
+                    .map(|point| 1.0 - line.distance_to_point(*point) / maximum_score_distance)
+                    .sum();
                 (line, score)
             })
-            .max_by_key(|scored_line| scored_line.1)
+            .max_by(|line_1, line_2| line_1.1.total_cmp(&line_2.1))
```

Comment:
since score is not a member of a line, maybe prefer pattern matching here (`(_, score_1), ...`)